tests/testthat/test-export_audit_graph_json.R

```diff
@@ -0,0 +1,34 @@
+test_that("basic functionality works as expected", {
+  all_flags <- c(
+    "Missing currency information",
+    "Negative or missing input value",
+    "Invalid or missing ISIN",
+    "Holding not in financial database",
+    "Included in analysis"
+  )
+
+  audit_file__ <- data.frame(
+    isin = c("ABC123", "DEF456", "GHI789", "JKL012", "MNO345"),
+    holding_id = 1:5,
+    flag = all_flags
+  )
+  export_path_full <- withr::local_tempdir()
+
+  export_audit_graph_json(audit_file__, paste0(export_path_full, "/coveragegraph"))
+
+  expected_output <- list(
+    key_1 = "Invalid input",
+    key_2 = "No data coverage",
+    key_3 = "Included in analysis"
+  )
+  output <- jsonlite::read_json(path = file.path(export_path_full, "coveragegraphlegend.json"))
+  expect_equal(output, expected_output)
+
+  expected_output <- list(
+    key_1 = 3,
+    key_2 = 1,
+    key_3 = 1
+  )
```

Comment:
I don't totally understand what is going on here

Reply:
The `coveragegraph.json` file is a specifically crafted JSON file that gives values to Constructiva to be reported back to the user in the webGUI. The values in `coveragegraph.json` match the keys in `coveragegraph.json`, so in this case there are:
1. 3 "Invalid input"
2. 1 "No data coverage"
3. 1 "Included in analysis"

which matches the expectation because the test specifically crafts an `audit_file__` that has those characteristics.